[CI] TransformInternalIndexIT.testUpdateDeletesOldTransformConfig

**Build scan**: https://gradle-enterprise.elastic.co/s/qvjg4l7qfqdxq 

**Repro line**: 
```
./gradlew ':x-pack:plugin:transform:internalClusterTest' --tests "org.elasticsearch.xpack.transform.integration.TransformInternalIndexIT.testUpdateDeletesOldTransformConfig" \
  -Dtests.seed=180F5806D3F6A363 \
  -Dtests.security.manager=true \
  -Dtests.locale=hi \
  -Dtests.timezone=Europe/Volgograd \
  -Druntime.java=11 \
  -Dtests.fips.enabled=true
```

**Reproduces locally?**: no

**Applicable branches**: `7.12`

**Failure history**:
<!--
Link to build stats and possible indication of when this started failing and how often it fails
<https://build-stats.elastic.co/app/kibana>
-->
**Failure excerpt**:

```
java.lang.AssertionError: test leaves indices that were not deleted: .transform-notifications-000002
Expected: []
     but: was [".transform-notifications-000002"]
	at __randomizedtesting.SeedInfo.seed([180F5806D3F6A363:75F6F23A53C81E91]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.elasticsearch.test.ESSingleNodeTestCase.tearDown(ESSingleNodeTestCase.java:134)
```

--- a/test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java
@@ -368,9 +368,6 @@ public abstract class ESSingleNodeTestCase extends ESTestCase {
             .actionGet();
 
         assertFalse("timed out waiting for shards to initialize", actionGet.isTimedOut());
-
-        // some tests leave with a non-green state
-        //assertThat(actionGet.getStatus(), equalTo(ClusterHealthStatus.GREEN));
     }
 
 }

--- a/test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java
@@ -114,6 +114,7 @@ public abstract class ESSingleNodeTestCase extends ESTestCase {
     @Override
     public void tearDown() throws Exception {
         logger.trace("[{}#{}]: cleaning up after test", getTestClass().getSimpleName(), getTestName());
+        ensureNoInitializingShards();
         super.tearDown();
         assertAcked(
             client().admin().indices().prepareDelete("*")
@@ -324,7 +325,6 @@ public abstract class ESSingleNodeTestCase extends ESTestCase {
         return ensureGreen(TimeValue.timeValueSeconds(30), indices);
     }
 
-
     /**
      * Ensures the cluster has a green state via the cluster health API. This method will also wait for relocations.
      * It is useful to ensure that all action on the cluster have finished and all shards that were currently relocating
@@ -353,6 +353,22 @@ public abstract class ESSingleNodeTestCase extends ESTestCase {
 
     protected boolean forbidPrivateIndexSettings() {
         return true;
+    }
+
+
+    /**
+     * waits until all shard initialization is completed.
+     *
+     * @throws IOException
+     */
+    protected void ensureNoInitializingShards() throws IOException {
+        ClusterHealthResponse actionGet = client().admin()
+            .cluster()
+            .health(Requests.clusterHealthRequest("_all").timeout(TimeValue.timeValueSeconds(70)).waitForNoInitializingShards(true))
+            .actionGet();
+
+        assertFalse("timed out waiting for shards to initialize", actionGet.isTimedOut());
+        assertThat(actionGet.getStatus(), equalTo(ClusterHealthStatus.GREEN));
     }
 
 }

--- a/test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java
@@ -41,6 +41,7 @@ import org.elasticsearch.node.NodeValidationException;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.script.MockScriptService;
 import org.elasticsearch.search.internal.SearchContext;
+import org.elasticsearch.test.rest.ESRestTestCase;
 import org.elasticsearch.transport.TransportSettings;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
@@ -359,12 +360,14 @@ public abstract class ESSingleNodeTestCase extends ESTestCase {
     /**
      * waits until all shard initialization is completed.
      *
+     * inspired by {@link ESRestTestCase}
+     *
      * @throws IOException
      */
     protected void ensureNoInitializingShards() throws IOException {
         ClusterHealthResponse actionGet = client().admin()
             .cluster()
-            .health(Requests.clusterHealthRequest("_all").timeout(TimeValue.timeValueSeconds(70)).waitForNoInitializingShards(true))
+            .health(Requests.clusterHealthRequest("_all").waitForNoInitializingShards(true))
             .actionGet();
 
         assertFalse("timed out waiting for shards to initialize", actionGet.isTimedOut());

--- a/test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java
@@ -368,7 +368,9 @@ public abstract class ESSingleNodeTestCase extends ESTestCase {
             .actionGet();
 
         assertFalse("timed out waiting for shards to initialize", actionGet.isTimedOut());
-        assertThat(actionGet.getStatus(), equalTo(ClusterHealthStatus.GREEN));
+
+        // some tests leave with a non-green state
+        //assertThat(actionGet.getStatus(), equalTo(ClusterHealthStatus.GREEN));
     }
 
 }

